GeoscienceAustralia/fc
# Test celery backend on real work loads

## Introduction

I have added extra parameters to `fc_app.py` to facilitate debugging

- `--skip-db-update`
- `--process-all` force processing for tiles even if they are already in the index, probably only safe to use in conjunction with `--skip-db-update` option.
- Added sample celery launcher script

I have been able to run fc app using celery backend from my test python environment. I needed to specify `--queue-size 100` parameter in order to avoid out of memory errors coming from Redis. This needs more investigation. What happens when celery can not submit more tasks because Redis is out memory? Can we deal with that situation gracefully (currently that's not happening)? What is sufficiently robust configuration we should use?  

## How to do test run 

1. Create `app.yaml` by copying one of the configs from `config` folder
2. Update `location` field to point to directory you can write to

Needs datacube install with `celery_runner` branch. To use my test environment you can try the following:
 
```
export PATH="/g/data/u46/users/kk7182/miniconda/bin:$PATH"
which activate
source activate agdc
#check that environment worked
datacube-fc --help
# submit for processing
qsub ./celery.sh
```
